### PR TITLE
#579 Adujust the height of the lineage iframe to fill the page

### PR DESCRIPTION
**Pull request by lokm01**

### CODE REVIEW

File: menas/ui/components/run/runDetail.view.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
             </Bar>
         </customHeader>
         <content>
-            <IconTabBar id="runIconTabBar" select="tabSelect">
+            <IconTabBar id="runIconTabBar" select="tabSelect" stretchContentHeight="true" expandable="false">
```
Comment (benedeki) [inline menas/ui/components/run/runDetail.view.xml]: Yeah that was the property I found about just before you mentioned you would take over. 😉

File: menas/ui/components/run/runDetail.controller.js [hunk 2]
```diff
@@ -35,6 +35,17 @@ sap.ui.define([
...
       this._detail = this.byId("detailPage");
       this._checkpointsTable = this.byId("Checkpoints");
+
+      this.byId("runIconTabBar").attachSelect(function(oSelectEv) {
+        if(oSelectEv.getParameter("selectedKey") === "lineage") {
+          const oFrame = $("#lineage_iframe");
+          oFrame.removeClass("lineageIframe");
+          setTimeout(function() {
+            //force re-render
+            oFrame.addClass("lineageIframe");
```
Comment (benedeki) [inline menas/ui/components/run/runDetail.controller.js]: Out of curiosity: do you know why the re-render is needed?

**Comment by benedeki** [comment]
Hmm, when I maximize the window or exit maximization, the `iframe` disappears until I resize the window by dragging the border or clicking the_Lineage_ button again. 😟 
(Windows 10.  Google Chrome5.0.3770.100 (Official Build) (64-bit) )

**Comment by GeorgiChochov** [comment]
> Hmm, when I maximize the window or exit maximization, the `iframe` disappears until I resize the window by dragging the border or clicking the_Lineage_ button again. 😟
> (Windows 10. Google Chrome5.0.3770.100 (Official Build) (64-bit) )

I can't reproduce this on a Mac.

**Comment by Zejnilovic** [comment]
Same as @benedeki 

Jul-09-2019 09-48-18

**Comment by GeorgiChochov** [comment]
This seems related to responsiveness, only happens when the navigation sidebar collapses or expands. Doesn't happen otherwise.

File: menas/ui/components/run/runDetail.controller.js (hunk 2 above)
Comment (lokm01) [reply menas/ui/components/run/runDetail.controller.js]: No idea. This whole thing is behaving strange.

**Comment by benedeki** [comment]
Just FYI, despite not supported by us, **Menas** seemed to work OK so far in Firefox (haven't tested deeply). But this _100% size_ is broken there - the `iframe` is hidden, probably height 0.

File: menas/ui/components/run/runDetail.controller.js [hunk 3]
```diff
@@ -36,18 +37,24 @@ sap.ui.define([
...
-          setTimeout(function() {
-            //force re-render
-            oFrame.addClass("lineageIframe");
-          }.bind(this), 200);
+          this.iframeHack();
         }
       }.bind(this));
     },
 
+    iframeHack: function() {
```
Comment (Zejnilovic) [inline menas/ui/components/run/runDetail.controller.js]: If it is a hack, maybe add a few comments to it so future generations know what was happening

**Pull request merged**